featurebyte/query_graph/node/function.py

```diff
@@ -0,0 +1,228 @@
+            column_input_args.extend(func_arg.get_column_args())
+        return column_input_args
+
+    @property
+    def max_input_count(self) -> int:
+        return len(self._get_column_function_args())
+
+    def _get_required_input_columns(
+        self, input_index: int, available_column_names: List[str]
+    ) -> Sequence[str]:
+        column_input_args = self._get_column_function_args()
+        if column_input_args[input_index] is None:
+            return []
+        return [column_input_args[input_index]]  # type: ignore
+
+    def derive_var_type(self, inputs: List[OperationStructure]) -> DBVarType:
+        return self.parameters.output_dtype
+
+    def _derive_node_operation_info(
+        self,
+        inputs: List[OperationStructure],
+        branch_state: OperationStructureBranchState,
+        global_state: OperationStructureInfo,
+    ) -> OperationStructure:
+        input_category = inputs[0].output_category
+        row_index_lineage = inputs[0].row_index_lineage
+        for input_operation_structure in inputs:
+            # check input category type is homogeneous
+            if input_operation_structure.output_category != input_category:
+                raise ValueError("Input category type is not homogeneous")
+            if input_operation_structure.output_type != NodeOutputType.SERIES:
+                raise ValueError("Input type is not series")
+            if input_operation_structure.row_index_lineage != row_index_lineage:
+                raise ValueError("Input row index is not matched")
+
+        # prepare node parameters
+        columns = []
+        aggregations = []
+        for input_operation_structure in inputs:
+            columns.extend(input_operation_structure.columns)
+            aggregations.extend(input_operation_structure.aggregations)
+
+        node_kwargs: Dict[str, Any] = {"columns": []}
+        if input_category == NodeOutputCategory.VIEW:
+            node_kwargs["columns"] = [
+                DerivedDataColumn.create(
+                    name=None,
+                    columns=columns,
+                    transform=self.parameters.function_name,
+                    node_name=self.name,
+                    dtype=self.parameters.output_dtype,
+                )
+            ]
+        else:
+            node_kwargs["columns"] = columns
+            node_kwargs["aggregations"] = [
+                PostAggregationColumn.create(
+                    name=None,
+                    columns=aggregations,
+                    transform=self.parameters.function_name,
+                    node_name=self.name,
+                    dtype=self.parameters.output_dtype,
+                )
+            ]
+
+        return OperationStructure(
+            **node_kwargs,
+            output_type=NodeOutputType.SERIES,
+            output_category=input_category,
+            row_index_lineage=row_index_lineage,
+        )
+
+    def _derive_sdk_code(
```

Comment:
Will test be added to cover feature definition generation?

Reply:
Tests are added in `tests/unit/query_graph/test_function.py`.
```
    # check the code generated by the generic function node
    state = SDKCodeExtractor(graph=global_graph).extract(node=gfunc)
    assert state.code_generator.statements[-1] == ("output", 'my_func(col, 1, 2.0, "hello", True)')
```